pkg/aws/policies.go

```diff
@@ -144,25 +146,30 @@ var roleTypeMap = map[string]string{
 
 func (c *awsClient) EnsureRole(name string, policy string, permissionsBoundary string,
 	version string, tagList map[string]string, path string, managedPolicies bool) (string, error) {
-	output, err := c.iamClient.GetRole(&iam.GetRoleInput{
+	output, err := c.iamClient.GetRole(context.Background(), &iam.GetRoleInput{
 		RoleName: aws.String(name),
 	})
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case iam.ErrCodeNoSuchEntityException:
-				return c.createRole(name, policy, permissionsBoundary, tagList, path)
-			default:
-				return "", err
-			}
+		if awserr.IsNoSuchEntityException(err) {
```

Comment:
Nice refactor! much better than the previous switch statement:)

Reply:
Thanks